packages/dds/tree/src/feature-libraries/object-forest/objectForest.ts

```diff
@@ -201,11 +201,11 @@ export class ObjectForest extends SimpleDependee implements IEditableForest {
     }
 
     private beforeChange(): void {
+        this.invalidateDependents();
```

Comment:
This must happen before `assert`, as there might be other trees (by other fluid clients) having active cursors. Although it does not prevent to apply a change to the "local" tree, but it prevents applying the change to those "remote" trees, when the "remote" SharedTree receives the change and tries to apply it to the Forest using the `ForestIndex`.

Reply:
This change is good. I originally intended this method to error if you had cursors when trying to edit: it wasn't designed with code like editable tree which lazily frees cursors in mind.